lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -762,90 +761,159 @@ Color _defaultGetDotColor(FlSpot _, double xPercentage, LineChartBarData bar) {
 }
 
 /// This class holds data about drawing spot dots on the drawing bar line.
 class FlDotData with EquatableMixin {
   /// Determines show or hide all dots.
   final bool show;
 
-  // Customizes the size of the dot
-  final double dotSize;
-
-  /// The stroke width to use for the corresponding [FlSpot]
-  final double strokeWidth;
-
-  /// Callback which is called to the the stroke color of the given [FlSpot].
-  /// The [FlSpot] is provided as parameter to this callback
-  final GetDotColorCallback getStrokeColor;
-
-  /// Callback which is called to the the color of the given [FlSpot].
-  /// The [FlSpot] is provided as parameter to this callback
-  final GetDotColorCallback getDotColor;
-
   /// Checks to show or hide an individual dot.
   final CheckToShowDot checkToShowDot;
 
+  /// Callback which is called to set the drawer of the given [FlSpot].
+  /// The [FlSpot] is provided as parameter to this callback
+  final GetDotDrawerCallback getDotDrawer;
+
   /// set [show] false to prevent dots from drawing,
-  /// [dotSize] determines the size of dots.
   /// if you want to show or hide dots in some spots,
   /// override [checkToShowDot] to handle it in your way.
-  ///
-  /// The color of dot determines by [getDotColor],
-  /// you can override it, it gives you a [FlSpot],
-  /// and you should decide to return a [Color].
-  ///
-  /// You can have stroke line around the dot,
-  /// by setting the thickness by [strokeWidth],
-  /// and you can implement [getStrokeColor] callback,
-  /// it gives you the [FlSpot], and you should decide to return a [Color].
   FlDotData({
     bool show,
-    double dotSize,
     CheckToShowDot checkToShowDot,
-    double strokeWidth,
-    GetDotColorCallback getStrokeColor,
-    GetDotColorCallback getDotColor,
+    GetDotDrawerCallback getDotDrawer,
   })  : show = show ?? true,
-        dotSize = dotSize ?? 4.0,
         checkToShowDot = checkToShowDot ?? showAllDots,
-        strokeWidth = strokeWidth ?? 0.0,
-        getStrokeColor = getStrokeColor ?? _defaultGetDotStrokeColor,
-        getDotColor = getDotColor ?? _defaultGetDotColor;
+        getDotDrawer = getDotDrawer ?? _defaultGetDotDrawer;
 
   /// Lerps a [FlDotData] based on [t] value, check [Tween.lerp].
   static FlDotData lerp(FlDotData a, FlDotData b, double t) {
     return FlDotData(
       show: b.show,
       checkToShowDot: b.checkToShowDot,
-      dotSize: lerpDouble(a.dotSize, b.dotSize, t),
-      strokeWidth: lerpDouble(a.strokeWidth, b.strokeWidth, t),
-      getDotColor: b.getDotColor,
-      getStrokeColor: b.getStrokeColor,
+      getDotDrawer: b.getDotDrawer,
     );
   }
 
   /// Used for equality check, see [EquatableMixin].
   @override
   List<Object> get props => [
         show,
-        dotSize,
-        strokeWidth,
-        getStrokeColor,
-        getDotColor,
         checkToShowDot,
+        getDotDrawer,
+      ];
+}
+
+abstract class FlDotDrawer with EquatableMixin {
```

Comment:
Hmm, let's talk about "FlDot**Drawer**" name,
Do you think it's descriptive and unique?
I think people will be confused about it, because we have a [Drawer](https://api.flutter.dev/flutter/material/Drawer-class.html) widget in the Flutter.
and on the other hand, we have CustomPainter that its function is about draw something.
In my opinion, **FlDotPainter** is more descriptive and better.
What do you think?

Reply:
I agree. I was actually thinking about that other Drawer myself